CMakeLists.txt

```diff
@@ -478,6 +478,25 @@ if ((NOT WITH_SYMENGINE_RCP) OR HAVE_TEUCHOS_BFD)
     set(WITH_SYMENGINE_TEUCHOS yes)
 endif()
 
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  ## References:
+  ## cmake  --help-policy CMP0042
+  set(CMAKE_MACOSX_RPATH ON)
+endif()
+
+if(CMAKE_INSTALL_RPATH_USE_LINK_PATH)
+  ## References:
+  ## https://cmake.org/Wiki/CMake_RPATH_handling#Mac_OS_X_and_the_RPATH
+  set(CMAKE_SKIP_BUILD_RPATH FALSE)
+  set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+  # set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+  list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
+  if("${isSystemDir}" STREQUAL "-1")
+     set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
```

Comment:
I'm not sure that this is a problem. `@rpath` always seems to get expanded to an absolute path:
```
cmake \
-DCMAKE_BUILD_TYPE=Debug \
-DINTEGER_CLASS:STRING=boostmp \
-DBoost_INCLUDE_DIR=$SPACK_VIEW_DIR/include \
-DBUILD_SHARED_LIBS:BOOL=on \
-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=on \
-DCMAKE_INSTALL_PREFIX=../install \  <--------
../symengine
```
still leads to 
```
$ otool -L ../install/lib/libsymengine.dylib
../install/lib/libsymengine.dylib:
	/Users/jp/Programs/GitHub/SymEngine/install/lib/libsymengine.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.1.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)

$ otool -l ../install/lib/libsymengine.dylib | grep LC_RPATH -A2
          cmd LC_RPATH
      cmdsize 64
         path /Users/jp/Programs/GitHub/SymEngine/install/lib (offset 12)
```
But, as I said, I'm really no `CMake`/build system expert. Would you mind providing some more detail as to exactly how you want me to change this line?